src/main/java/org/opensearch/knn/common/launchcontrol/KNNLaunchControl.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.common.launchcontrol;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.knn.index.KNNSettings;
+
+import java.util.stream.Stream;
+
+public final class KNNLaunchControl {
+
+    // Feature flags
+    public static final String KNN_LAUNCH_QUERY_REWRITE_ENABLED = "knn.launch.query.rewrite.enabled";
+    private static final boolean KNN_LAUNCH_QUERY_REWRITE_ENABLED_DEFAULT = true;
+
+    public static Setting<Boolean> KNN_LAUNCH_QUERY_REWRITE_ENABLED_SETTING = Setting.boolSetting(
+        KNN_LAUNCH_QUERY_REWRITE_ENABLED,
+        KNN_LAUNCH_QUERY_REWRITE_ENABLED_DEFAULT,
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
+
+    public static Stream<Setting<?>> getFeatureFlags() {
+        return Stream.of(KNN_LAUNCH_QUERY_REWRITE_ENABLED_SETTING);
+    }
+
+    public static boolean isFeatureEnabled(final String featureName) {
+        return Boolean.parseBoolean(KNNSettings.state().getSettingValue(featureName).toString());
+    }
```

Comment:
rather than having such generic function, I would recommend moving to isFeatureABCEnabled(). 
1. This ensures client doesn't need to look for string is required to be passed for this feature flag.
2. If this logic of enabling/disabling a feature become complex we just have to make change at 1 place.

Reply:
It feels like a wrapper which we can get away with, I don't have a preference here

Reply:
Lets put a wrapper. Lets make it easy for the caller.